c/misra/src/rules/RULE-9-7/UninitializedAtomicObject.ql

```diff
@@ -0,0 +1,75 @@
+/**
+ * @id c/misra/uninitialized-atomic-object
+ * @name RULE-9-7: Atomic objects shall be appropriately initialized before being accessed
+ * @description Atomic objects that do not have static storage duration shall be initialized with a
+ *              value or by using 'atomic_init()'.
+ * @kind problem
+ * @precision high
+ * @problem.severity warning
+ * @tags external/misra/id/rule-9-7
+ *       concurrency
+ *       external/misra/c/2012/amendment4
+ *       external/misra/obligation/mandatory
+ */
+
+import cpp
+import codingstandards.c.misra
+import semmle.code.cpp.controlflow.Dominance
+
+class ThreadSpawningFunction extends Function {
+  ThreadSpawningFunction() {
+    this.hasName("pthread_create")
+    or
+    this.hasName("thrd_create")
+    or
+    exists(FunctionCall fc |
+      fc.getTarget() instanceof ThreadSpawningFunction and
+      fc.getEnclosingFunction() = this
+    )
+  }
+}
+
+class AtomicInitAddressOfExpr extends FunctionCall {
+  Expr addressedExpr;
+
+  AtomicInitAddressOfExpr() {
+    exists(AddressOfExpr addrOf |
+      getArgument(0) = addrOf and
+      addrOf.getOperand() = addressedExpr and
+      getTarget().getName() = "__c11_atomic_init"
```

Comment:
Is this name required by the standard? If not, we need to confirm it's used by all our supported compilers, or use a different approach (identify the function call wrapped by the macro `atomic_init`?)

Reply:
OK, I quickly authored a little library to make this easier going forward.